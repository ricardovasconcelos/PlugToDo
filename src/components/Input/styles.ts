import styled from 'styled-components';

export const InputComponent = styled.input`
  border: 1px solid ${(props) => props.theme['gray-500']};
  height: 3rem;
  border-radius: 4px;
  padding: 1rem;
  line-height: 150%;
  width: 100%;

  ::placeholder {
    font-weight: bold;
    color: ${(props) => props.theme['gray-800']};
  }

  &:focus {
    border: 0;
  }
`;
