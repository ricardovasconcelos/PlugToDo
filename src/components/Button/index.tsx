import { ButtonHTMLAttributes, ReactNode } from 'react';

import { ButtonComponent } from './styles';

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  children: ReactNode;
  width: number;
}

export function Button({ children, ...props }: ButtonProps) {
  return <ButtonComponent {...props}>{children}</ButtonComponent>;
}
