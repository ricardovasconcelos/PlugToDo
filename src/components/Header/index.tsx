import { HeaderContainer } from './styles';
import logo from '../../assets/logo.svg';
import { Link } from 'react-router-dom';

export function Header() {
  return (
    <HeaderContainer>
      <Link to='/'>
        <img src={logo} alt='Plug Logo' />
      </Link>
    </HeaderContainer>
  );
}
