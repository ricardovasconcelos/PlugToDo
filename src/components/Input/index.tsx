import { InputHTMLAttributes } from 'react';

import { InputComponent } from './styles';

interface InputProps extends InputHTMLAttributes<HTMLInputElement> {}

export function Input({ ...props }: InputProps) {
  return <InputComponent {...props} />;
}
