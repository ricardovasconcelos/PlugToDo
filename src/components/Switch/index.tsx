import { InputHTMLAttributes } from 'react';
import { SwithContainer, SwitchWrapper, SwitchCheckbox, SwitchItem } from './styles';

interface SwitchProps extends InputHTMLAttributes<HTMLInputElement> {
  label: string;
  checked?: boolean;
}

export function Switch({ label, checked, ...props }: SwitchProps) {
  return (
    <SwithContainer data-testid='switch-container'>
      <p>{label}</p>

      <SwitchWrapper>
        <SwitchCheckbox id='checkbox' type='checkbox' checked={checked} {...props} />
        <SwitchItem htmlFor='checkbox' />
      </SwitchWrapper>
    </SwithContainer>
  );
}
