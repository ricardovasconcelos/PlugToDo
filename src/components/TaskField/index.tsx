import { TaskFieldContainer, TaskDetailContainer } from './styles';
import { Task } from '../../db/db.types';

interface TaskFieldProps extends Omit<Task, 'id'> {
  onClickTask: () => void;
}

export function TaskField({ title, description, done, onClickTask }: TaskFieldProps) {
  return (
    <TaskFieldContainer done={done} onClick={onClickTask}>
      <TaskDetailContainer>
        <h3>{title}</h3>
        <p>{description}</p>
      </TaskDetailContainer>
    </TaskFieldContainer>
  );
}
