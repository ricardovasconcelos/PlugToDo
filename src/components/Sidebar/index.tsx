import { useRef, useEffect, useCallback, MouseEvent } from 'react';
import { useSpring, animated } from 'react-spring';
import { Button } from '../Button';
import { Input } from '../Input';

import {
  SidebarContainer,
  SidebarWrapper,
  SidebarContent,
  SidebarHeader,
  CloseSidebar,
  SidebarFooter,
} from './styles';

import close from '../../assets/close.svg';

interface SidebarProps {
  showSidebar: boolean;
  onShowSidebar: (value: boolean) => void;
}

export const Sidebar = ({ showSidebar, onShowSidebar }: SidebarProps) => {
  const sidebarRef = useRef<HTMLDivElement>(null);

  const animation = useSpring({
    config: {
      duration: 250,
    },
    opacity: showSidebar ? 1 : 0,
    transform: showSidebar ? `translateY(0%)` : `translateY(-100%)`,
  });

  const closeSidebar = (event: MouseEvent<HTMLDivElement>) => {
    if (sidebarRef.current === event.target) {
      onShowSidebar(false);
    }
  };

  const keyPress = useCallback(
    (event: KeyboardEvent) => {
      if (event.key === 'Escape' && showSidebar) {
        onShowSidebar(false);
      }
    },
    [onShowSidebar, showSidebar],
  );

  useEffect(() => {
    document.addEventListener('keydown', keyPress);
    return () => document.removeEventListener('keydown', keyPress);
  }, [keyPress]);

  return (
    <>
      {showSidebar ? (
        <SidebarContainer onClick={closeSidebar} ref={sidebarRef}>
          <animated.div style={animation}>
            <SidebarWrapper>
              <SidebarHeader>
                <CloseSidebar onClick={() => onShowSidebar(false)}>
                  <img src={close} alt='Botão para fechar modal de tarefas' />
                </CloseSidebar>
                <p>Nova Tarefa</p>
              </SidebarHeader>

              <SidebarContent>
                <Input placeholder='Título' />

                <Input placeholder='Descrição' />
              </SidebarContent>

              <SidebarFooter>
                <Button width={20}>CRIAR TAREFA</Button>
              </SidebarFooter>
            </SidebarWrapper>
          </animated.div>
        </SidebarContainer>
      ) : null}
    </>
  );
};
