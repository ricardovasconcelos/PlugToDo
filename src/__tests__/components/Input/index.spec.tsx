import { screen, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Input } from '../../../components/Input';

describe('Input', () => {
  it('should be able to render a Input', () => {
    const { container } = render(<Input placeholder='input testing' />);
    const input = screen.getByPlaceholderText('input testing');
    expect(input).toHaveStyle({
      height: '3rem',
      borderRadius: '4px',
      padding: '1rem',
      lineHeight: '150%',
      width: '100%',
    });

    expect(input).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Changes its value when typing', async () => {
    render(<Input placeholder='input testing' />);

    const input = screen.getByPlaceholderText('input testing');
    const text = 'This is my new text';
    userEvent.type(input, text);

    await waitFor(() => {
      expect(input).toHaveValue(text);
    });
  });
});
