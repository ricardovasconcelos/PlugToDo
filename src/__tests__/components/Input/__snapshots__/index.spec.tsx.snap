// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should be able to render a Input 1`] = `
.c0 {
  border: 1px solid;
  height: 3rem;
  border-radius: 4px;
  padding: 1rem;
  line-height: 150%;
  width: 100%;
}

.c0::-webkit-input-placeholder {
  font-weight: bold;
}

.c0::-moz-placeholder {
  font-weight: bold;
}

.c0:-ms-input-placeholder {
  font-weight: bold;
}

.c0::placeholder {
  font-weight: bold;
}

.c0:focus {
  border: 0;
}

<input
  class="c0"
  placeholder="input testing"
/>
`;
